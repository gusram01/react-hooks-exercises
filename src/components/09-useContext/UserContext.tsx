import { createContext, Dispatch, SetStateAction } from "react";

export interface User {
  id?: number;
  name?: string;
  email?: string;
}

export interface Context {

  user?: User;
  setUser?: Dispatch<SetStateAction<User>>;

}

let some: Context = {};

export const UserContext = createContext(some);