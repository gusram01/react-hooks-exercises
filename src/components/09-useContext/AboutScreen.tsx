import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const AboutScreen = () => {

  const { user, setUser } = useContext(UserContext);

  const handleCLick = () => setUser!({});


  return (
    <>
      <h1>About Page</h1>
      <hr />


      <pre>
        {JSON.stringify(user, null, 3)}
      </pre>
      <button
        className="btn btn-warning"
        onClick={handleCLick}
      >
        LogOut
      </button>
    </>
  );
};

export default AboutScreen;
