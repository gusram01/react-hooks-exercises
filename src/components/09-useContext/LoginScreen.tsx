import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const LoginScreen = () => {

  const { user, setUser } = useContext(UserContext);
  const dummyUser = {
    id: 101023,
    name: 'Gus'
  };
  return (
    <>
      <h1>Login Page</h1>
      <hr />

      <button
        className="btn btn-primary"
        onClick={() => setUser!(dummyUser)}
      >
        Login
      </button>

      <pre>
        {JSON.stringify(user, null, 3)}
      </pre>
    </>
  );
};

export default LoginScreen;
