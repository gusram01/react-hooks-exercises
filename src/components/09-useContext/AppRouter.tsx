import React from 'react'
import AboutScreen from './AboutScreen';
import LoginScreen from './LoginScreen';
import HomeScreen from './HomeScreen';
import NavBar from './NavBar';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";

const AppRouter = () => {
  return (
    <Router>

      <div>
        <NavBar />

        <div className="container">
          <Switch>
            <Route exact path="/" component={HomeScreen} />
            <Route exact path="/about" component={AboutScreen} />
            <Route exact path="/login" component={LoginScreen} />
            <Redirect to="/" />
          </Switch>
        </div>

      </div>

    </Router>
  )
}

export default AppRouter;