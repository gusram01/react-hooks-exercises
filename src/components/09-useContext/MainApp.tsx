import React, { useState } from 'react';
import AppRouter from './AppRouter';
import { UserContext } from './UserContext';

const MainApp = () => {

  // const user = {
  //   id: 1234,
  //   name: 'Gus',
  //   email: 'd@d.com'
  // }
  const [user, setUser] = useState({});

  return (
    <UserContext.Provider value={{ user, setUser }}>
      <AppRouter />
    </UserContext.Provider>
  )
};

export default MainApp;