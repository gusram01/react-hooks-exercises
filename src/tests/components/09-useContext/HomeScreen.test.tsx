import React from 'react';
import { shallow, mount } from 'enzyme';
import HomeScreen from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';
import { User, Context } from '../../../components/09-useContext/UserContext';

describe('HomeScreen tests', () => {

  const user: User = {
    id: 1,
    name: 'gus',
    email: 'g@g.com',
  }

  const context: Context = {
    user
  }

  console.log(user);
  const wrapper = mount(
    <UserContext.Provider value={context} >
      <HomeScreen />
    </UserContext.Provider >
  );
  test('HomeScree should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
