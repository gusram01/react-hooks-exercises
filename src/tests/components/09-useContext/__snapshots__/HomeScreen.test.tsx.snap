// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen tests HomeScree should render correctly 1`] = `
Array [
  <h1>
    Home Page
  </h1>,
  <hr />,
  <pre>
    {
   "id": 1,
   "name": "gus",
   "email": "g@g.com"
}
  </pre>,
]
`;
